src/qibo/backends/__init__.py

```diff
@@ -132,18 +144,25 @@ def __repr__(self):
     def show_config(self):
         log.info(f"Using {self} backend on {self.active_backend.default_device}")
 
-    def check_availability(self, module_name):
+    def check_availability(self, module_name, check_version=True):
         """Check if module is installed.
 
         Args:
             module_name (str): module name.
+            check_version (str): check if minimum version.
 
         Returns:
             ``True`` if the module is installed, ``False`` otherwise.
         """
         if module_name not in self._availability:
             from pkgutil import iter_modules
+            from importlib_metadata import version
             is_available = module_name in (name for _, name, _ in iter_modules())
+            if is_available and check_version:
+                minimum_version = self._backends_min_version.get(module_name)
+                if version(module_name) < minimum_version: # pragma: no cover
```

Comment:
In principle the `minimum_version` entry in profiles.yml is optional and the user can add a backend there without specifiying this, for example if any version of this backend is supported by qibo. Should we add an exception to the version check in this case? I believe
```suggestion
                if minimum_version is not None and version(module_name) < minimum_version: # pragma: no cover
```
would work.

Reply:
Thanks. I have also included the versioning parser from packaging because the string comparison doesn't work always.